tests/rptest/tests/alter_topic_configuration_test.py

```diff
@@ -323,3 +353,174 @@ def check():
+            }, {
+                'redpanda.remote.read': 'true',
+                'redpanda.remote.write': 'false'
+            }]
+            for props in props_list:
+                kcl.alter_topic_config(props,
+                                       incremental_update,
+                                       topic,
+                                       node=self.redpanda.controller())
+                wait_until(lambda: func(props) == True,
+                           timeout_sec=10,
+                           backoff_sec=1,
+                           err_msg=f'Failed check {func.__name__}')
+
+        def restart_node_and_await_stable_leader(node):
+            should_sleep = node == self.redpanda.controller()
+            self.redpanda.restart_nodes(node)
+            self.redpanda.wait_for_membership(first_start=False)
+            self.redpanda._admin.await_stable_leader(namespace='redpanda',
+                                                     topic='controller',
+                                                     partition=0)
+            if should_sleep:
+                # This is a hack to get around controller node stabilization post restart.
+                time.sleep(2)
+
+        # Perform alterations and consistency checks across cluster with all un-upgraded nodes
+        alter_and_check(check_consistent_properties_across_nodes)
+
+        # Install updated version of redpanda across all nodes.
+        self.redpanda._installer.install(self.redpanda.nodes,
+                                         RedpandaInstaller.HEAD)
+        # Upgrade one node.
+        restart_node_and_await_stable_leader(self.redpanda.nodes[0])
+
+        # We can now expect to see this log line on the first node during an `AlterConfig` request.
+        log_line = 'Performing deprecated incremental_update to shadow_indexing_mode'
+        # Set log level to trace for cluster, since that's where log_line appears.
+        self.redpanda._admin.set_log_level('cluster', 'trace')
+
+        # Perform alterations and consistency checks across cluster with only one upgraded node
+        alter_and_check(check_consistent_properties_across_nodes)
+
+        # Check for deprecated path log line on the upgraded node.
+        assert self.redpanda.search_log_node(self.redpanda.nodes[0],
+                                             pattern=log_line) == True
+
+        # Restart the second node.
+        restart_node_and_await_stable_leader(self.redpanda.nodes[1])
+
+        # Set the log level again for restarted node.
+        self.redpanda._admin.set_log_level('cluster', 'trace')
+
+        # Perform alterations and consistency checks across cluster with two upgraded nodes
+        alter_and_check(check_consistent_properties_across_nodes)
+
+        # Check for deprecated path log line on the upgraded nodes.
+        assert self.redpanda.search_log_node(self.redpanda.nodes[0],
+                                             pattern=log_line) == True
```

Comment:
Yeah, you're right about this. Unfortunately it seems like `trim_log()` and `search_log_node()` are somewhat race-y around the second topic update.

I have added new assertions around the feature flag being `unavailable` and `active` that should cover the same code path as the deprecated log line, and changed the check on the second property update to only check the newly upgraded node instead of both, due to the unexpected behavior.